src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -136,18 +136,32 @@ trait WorkflowComponent {
 
         insertInBatches(submissionValidationQuery, inputResolutionRecs) andThen
         insertedRecQuery.result.map(_.map { case (workflowEntityRec, insertedInputResolutionRec) =>
-          (AttributeEntityReference(workflowEntityRec.entityType, workflowEntityRec.name), insertedInputResolutionRec.inputName) -> insertedInputResolutionRec
+          val ref = AttributeEntityReference(workflowEntityRec.entityType, workflowEntityRec.name)
+          val name = insertedInputResolutionRec.inputName
+          (ref, name) -> insertedInputResolutionRec
         }.toMap)
       }
 
-      def insertInputResolutionAttributes(workflows: Seq[Workflow], inputResolutionRecs: Map[(AttributeEntityReference, String), SubmissionValidationRecord]) = {
-        val attributeRecs = for {
+      def insertInputResolutionAttributes(workflows: Seq[Workflow], inputResolutionRecs: Map[(AttributeEntityReference, String), SubmissionValidationRecord]): ReadWriteAction[Unit] = {
+        val attributes = for {
           workflow <- workflows
           inputResolution <- workflow.inputResolutions
           attribute <- inputResolution.value
-        } yield submissionAttributeQuery.marshalAttribute(inputResolutionRecs(workflow.workflowEntity, inputResolution.inputName).id, inputResolution.inputName, attribute, Map.empty)
+        } yield (workflow, inputResolution, attribute)
 
-        submissionAttributeQuery.batchInsertAttributes(attributeRecs.flatten)
+        // each attribute may be marshalled into one or more records
+        // so including the marshal step in the above yield would result in a Seq[Seq[]]
+
+        val attributeRecs = attributes flatMap { case (workflow, inputResolution, attribute) =>
+          val ownerId = inputResolutionRecs(workflow.workflowEntity, inputResolution.inputName).id
+
+          // TODO fix this!
```

Comment:
they are stored as attributes because they can be different types and the attribute system works with types nicely. Another reason this makes sense is that the source of the resolutions are attributes so it makes sense to store them the same way. But I agree that namespace does not make much sense here... unless we want to do things like differentiate regular inputs from runtime, etc.